File: app/Facts/shared/src/commonMain/kotlin/com/usefulapps/facts/data/model/Result.kt
package com.usefulapps.facts.data.model

import io.ktor.http.Url
import kotlinx.serialization.Serializable

@Serializable
data class Result(
    val found: Boolean,
    val site: String,
    val url: String,
    val snippet: String,
    val numbers_matched: Int,
    val phrases_matched: Int,
    val verdict: Boolean,
    val error: String? = null
)

File: engine/model/result.py
from pydantic import BaseModel
from typing import Optional

class Result(BaseModel):
    found: bool = False
    site: str = ""
    url: str = ""
    snippet: str = ""
    numbers_matched: int = -1
    phrases_matched: int = -1
    verdict: bool = False
    error: Optional[str] = None
